JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -17,6 +17,8 @@ module Microsoft.ApplicationInsights {
         accountId: () => string;
         sessionRenewalMs: () => number;
         sessionExpirationMs: () => number;
+        properties: () => Object;
+        measurements: () => Object;
```

Comment:
@AlexBulankou I'm ok with removing config.properties approach that i've implemented in my 1st iteration
@olegsych I like onBeforeSendTelemetry, even though i'm not sure we need an interface here. The 2nd one (config.properties[xxx] = () => {return sometihng}) looks very good. However its limitation is that if you need to set several properties you end up writing unnecessary code (comparing to onBeforeSendTelemetry).